extractor.go

```diff
@@ -158,7 +158,7 @@ func (te *Extractor) Extract(reader io.Reader) error {
 		// Checks if the relative path matches or exceeds the root
 		// We check for matching because the outputPath function strips the original root
 		rel, err := fp.Rel(rootOutputPath, outputPath)
-		if err != nil || rel == "." || strings.Contains(rel, "..") {
+		if err != nil || rel == "." {
```

Comment:
This fix might be too permissive. IIRC the point of this check was essentially a fallback to make sure that given two OS paths that the output path is not a parent of the root path (excluding any symlink shenanigans since we won't traverse them anyway).

cc @Stebalien

Reply:
Yeah, I think we need to change this to ensure that no path component is ".." exactly.